iModelJsNodeAddon/api_package/ts/src/NativeLibrary.ts

```diff
@@ -92,7 +92,12 @@ export class NativeLibrary {
   public static load() {
     if (!this._nativeLib) {
       try {
-        this._nativeLib = require(`./${NativeLibrary.archName}/${NativeLibrary.nodeAddonName}`) as typeof IModelJsNative; // eslint-disable-line @typescript-eslint/no-var-requires
+        const platform = os.platform() as NodeJS.Platform | "ios"; // we add "ios"

```

Comment:
My changes here definitely work for loading the addon right now.
For the other two, (bufferutil and utf-8-validate), I guess we have to patch the node js like you did.

Reply:
I don't think NativeLibrary.ts should be papering over the fact that the mobile addon is linked into node on that platform.
We don't have a choice, obviously, for the other two...

Reply:
I added your override logic to the internal PR: https://github.com/iTwin/imodel-native-internal/pull/133/commits/4ee2c08903ac54673c8a779f393a5b7e975b8447

Reply:
@khanaffan @pmconne @aruniverse Do any of you agree that there is value in highlighting the linkedbinding call for mobile in NativeLibrary like this?

I think so, but it is technically not needed anymore now that I have Affan's patch back into the new node source.

Reply:
My patch is already part of the code. I was just wondering why it stopped working. Should we investigate fixing that instead of having to have a patch on top of it for the exact same problem? You may alternatively remove the following just to make sure it's not confusing how we load iModelJsNative binding.

```js
    } else if (name.endsWith("imodeljs.node")) {
      return process._linkedBinding("iModelJsNative");
    }
```

Reply:
> My patch is already part of the code. I was just wondering why it stopped working. Should we investigate fixing that instead of having to have a patch on top of it for the exact same problem? You may alternatively remove the following just to make sure it's not confusing how we load iModelJsNative binding.
> 
> ```js
>     } else if (name.endsWith("imodeljs.node")) {
>       return process._linkedBinding("iModelJsNative");
>     }
> ```

Your patch went away when I upgrade all the node source to v18 (in the internal PR that is paired with this one).

Reply:
> You may alternatively remove the following just to make sure it's not confusing how we load iModelJsNative binding.

@khanaffan I did that: https://github.com/iTwin/imodel-native-internal/pull/133/commits/1e94a655570fa0623e4e20986195b3cc8682a11e

So, now we have require patched to load our builds of bufferutil and utf-8-validate and the mobile addon is explicitly loaded in NativeLibrary.ts